File: src/main/java/site/gachontable/gachontablebe/domain/waiting/presentation/dto/response/WaitingHistoryResponse.java
package site.gachontable.gachontablebe.domain.waiting.presentation.dto.response;

import lombok.Builder;
import site.gachontable.gachontablebe.domain.waiting.domain.Waiting;
import site.gachontable.gachontablebe.domain.waiting.type.Status;

import java.time.LocalDateTime;
import java.util.UUID;

import static site.gachontable.gachontablebe.domain.shared.DateTimeFormatters.WITH_WEEKDAY;

@Builder
public record WaitingHistoryResponse(
        UUID waitingId,
        String pubName,
        Status status,
        String enteredTime,
        String exitTime) {

    public static WaitingHistoryResponse of(Waiting waiting, LocalDateTime exitTime) {

        return WaitingHistoryResponse.builder()
                .waitingId(waiting.getWaitingId())
                .pubName(waiting.getPub().getPubName())
                .status(waiting.getWaitingStatus())
                .enteredTime(waiting.getUpdatedAt().format(WITH_WEEKDAY))
                .exitTime(exitTime != null ? exitTime.format(WITH_WEEKDAY) : null)
                .build();
    }
}

File: src/main/kotlin/site/gachontable/presentation/waiting/dto/response/WaitingHistoryResponse.kt
package site.gachontable.presentation.waiting.dto.response

import site.gachontable.domain.waiting.domain.Waiting
import site.gachontable.domain.waiting.type.Status
import site.gachontable.presentation.shared.DateTimeFormatters
import java.time.LocalDateTime
import java.util.UUID

data class WaitingHistoryResponse(
    val waitingId: UUID,
    val pubName: String,
    val status: Status,
    val enteredTime: String,
    val exitTime: String,
) {
    companion object {
        fun of(waiting: Waiting, exitTime: LocalDateTime?): WaitingHistoryResponse =
            WaitingHistoryResponse(
                waitingId = waiting.id!!,
                pubName = waiting.pub.pubName,
                status = waiting.waitingStatus,
                enteredTime = waiting.updatedAt!!.format(DateTimeFormatters.WITH_WEEKDAY),
                exitTime = exitTime?.format(DateTimeFormatters.WITH_WEEKDAY).toString()
            )
    }
}